### PR TITLE
Generalize pto constraint tracking for multiple heaps in sep theory

### PR DESCRIPTION
**Pull request by ajreynol**
Fixes https://github.com/cvc5/cvc5/issues/8659.

### CODE REVIEW

File: src/theory/sep/theory_sep.cpp [hunk 1]
```diff
@@ -1744,105 +1747,128 @@ bool TheorySep::areDisequal( Node a, Node b ){
...
-        if (satom.getKind() == SEP_PTO)
+        // two positive pto
+        if (!areEqual(pval, qval))
         {
-          if( areEqual( atom[1], ei_n ) ){
-            addPto( ei, ei_n, atom, false );
+          std::vector<Node> exp;
+          if (plbl != qlbl)
+          {
+            Assert(areEqual(plbl, qlbl));
```
Comment (mudathirmahgoub) [inline src/theory/sep/theory_sep.cpp:1818]: Can you say here why these labels are equal ?

File: src/theory/sep/theory_sep.cpp [hunk 2]
```diff
@@ -1744,105 +1747,128 @@ bool TheorySep::areDisequal( Node a, Node b ){
...
+            exp.push_back(plbl.eqNode(qlbl));
+          }
+          exp.push_back(p);
+          exp.push_back(q);
+          // enforces injectiveness of pto
+          //  (pto x y) ^ (pto y w) ^ x = y => y = w
```
Comment (mudathirmahgoub) [inline src/theory/sep/theory_sep.cpp:1824]: My understanding locations and values can have different types. So `y` here appears as a value in `(pto x y)` and as a location in `(pto y w)`. Is this a typo?

Comment (mudathirmahgoub) [reply src/theory/sep/theory_sep.cpp:1824]: Also, the premise `x = y` here equates between locations, but the code equates between labels `plbl = qlbl` which are sets.

File: src/theory/sep/theory_sep.cpp [hunk 3]
```diff
@@ -1744,105 +1747,128 @@ bool TheorySep::areDisequal( Node a, Node b ){
...
+          exp.push_back(neg.notNode());
+          std::vector<Node> conc;
+          if (pval != qval)
+          {
+            conc.push_back(pval.eqNode(qval).notNode());
+          }
+          Node concn = nm->mkOr(conc);
+          Trace("sep-pto") << "prop neg/pos: " << concn << " by " << exp
+                           << std::endl;
+          // propagation (pto x y) ^ ~(pto z w) ^ x = z => y != w
```
Comment (mudathirmahgoub) [inline src/theory/sep/theory_sep.cpp:1860]: Same issue here. The comment uses  equality between locations `x, z` but the code uses equality between sets `plbl, qlbl`

**Comment by mudathirmahgoub** [comment]
Unrelated to this PR, but in this file
https://raw.githubusercontent.com/cvc5/cvc5/main/docs/theories/separation-logic.rst
There is a typo in line 25 which says `... the operator ``sep`` denotes separation start`
It should be `separation star`

File: src/theory/sep/theory_sep.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/sep/theory_sep.cpp:1824]: Yes this is a typo, I'll fix.

**Pull request merged**